app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -1,17 +1,65 @@
-// package ch.epfl.favo.common;
-//
-//         import com.google.firebase.firestore.FirebaseFirestore;
-//
-// public final class DatabaseWrapper {
-//
-//     private static FirebaseFirestore mFirestore;
-//
-//     public DatabaseWrapper() {
-//         FirebaseFirestore.setLoggingEnabled(true);
-//         mFirestore = FirebaseFirestore.getInstance();
-//     }
-//
-//     public static FirebaseFirestore getDatabaseInstance() {
-//         return mFirestore;
-//     }
-// }
+package ch.epfl.favo.common;
+
+import android.util.Log;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
+
+import java.util.Map;
+import java.util.Random;
+
+public class DatabaseWrapper {
+
+    private static final String TAG = "DatabaseWrapper";
+    private static DatabaseWrapper INSTANCE = null;
+    private FirebaseFirestore firestore;
+
+    // final fields regarding ID generation
+    private static String ID_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+    private static final int ID_LENGTH = 25;
+
+    private DatabaseWrapper() {
+        FirebaseFirestore.setLoggingEnabled(true);
+        FirebaseFirestoreSettings settings =
+                new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(true).build();
+        try {
+            firestore = FirebaseFirestore.getInstance();
+        } catch (Exception e) {
+            Log.d(TAG, e.toString());
+        }
+        firestore.setFirestoreSettings(settings);
```

Comment:
If line 27 throws, then `firestore` is an un-initialized object.

Reply:
you are right good sir